ipn/ipnlocal/local.go

```diff
@@ -2064,6 +2084,40 @@ func (b *LocalBackend) updateFilterLocked(netMap *netmap.NetworkMap, prefs ipn.P
 	}
 }
 
+// captivePortalWarnable is a Warnable which is set to an unhealthy state when a captive portal is detected.
+var captivePortalWarnable = health.Register(&health.Warnable{
+	Code:  "captive-portal-detected",
+	Title: "Captive portal detected",
+	// High severity, because captive portals block all traffic and require user intervention.
+	Severity:            health.SeverityHigh,
+	Text:                health.StaticMessage("This Wi-Fi network requires you to log in manually."),
```

Comment:
sure it's Wi-Fi? You filtering that somewhere? I don't know how we can detect that in general.